### PR TITLE
v1.0 Release

### PR DESCRIPTION
**Pull request by postatum**

### CODE REVIEW

File: src/js2dt.js [hunk 1]
```diff
@@ -0,0 +1,352 @@
...
+  }
+
+  /**
+   * Process definitions the same way main type is processed
+   * (using ramlForm).
+   *
+   * @param  {Object} defs - JSON 'definitions' object.
+   * @returns  {Object}
+   */
+  this._processDefinitions = function (defs) {
```
Comment (jstoiko) [inline src/js2dt.js]: or why not the `_` prefix on the other ones below

File: src/utils.js [hunk 2]
```diff
@@ -0,0 +1,51 @@
+'use strict'
+
+/**
+ * Merge second object into first one.
+ *
+ * @param  {Object} obj
+ * @param  {Object} upd
+ * @returns  {Object}
+ */
+function updateObjWith (obj, upd) {
```
Comment (jstoiko) [inline src/utils.js]: forget this.

**Comment by postatum** [comment]
> some integration tests are failing

Yes, because they test features that are not implemented yet.

File: src/utils.js [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+  return title(name)
+}
+
+/**
+ * Clone object.
+ *
+ * @param  {Object} obj
+ * @returns  {Object}
+ */
+function cloneObj (obj) {
```
Comment (postatum) [inline src/utils.js]: Removed

File: examples/dt2js_example.js [hunk 4]
```diff
@@ -0,0 +1,10 @@
+'use strict'
+var raml2json = require('ramldt2jsonschema')
+
```
Comment (postatum) [inline examples/dt2js_example.js]: Fixed  https://github.com/raml-org/ramldt2jsonschema/commit/cdd565d8f005a61a701dca2d1869da79fff9e7dc

File: src/dt2js.js [hunk 5]
```diff
@@ -0,0 +1,228 @@
...
+var constants = require('./constants')
+var utils = require('./utils')
+
+/**
+ * Get RAML Data Types context.
+ *
+ * @param  {string} fileName - File from which to get context.
+ * @returns  {Object} - RAML data types context.
+ */
+function getRAMLContext (fileName) {
```
Comment (postatum) [inline src/dt2js.js]: Fixed https://github.com/raml-org/ramldt2jsonschema/commit/cdd565d8f005a61a701dca2d1869da79fff9e7dc

File: src/dt2js.js [hunk 6]
```diff
@@ -0,0 +1,228 @@
...
+ */
+
+/**
+ * Convert RAML data type to JSON schema.
+ *
+ * @param  {string} fileName - File in which type is located.
+ * @param  {string} typeName - Name of the type to be converted.
+ * @param  {conversionCallback} cb - Callback to be called with converted value.
+ */
+function dt2js (fileName, typeName, cb) {
```
Comment (postatum) [inline src/dt2js.js]: Fixed https://github.com/raml-org/ramldt2jsonschema/commit/cdd565d8f005a61a701dca2d1869da79fff9e7dc

File: src/dt2js_cli.js [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+}
+
+/**
+ * Just call dt2js.
+ *
+ * @param  {string} ramlFile
+ * @param  {string} ramlTypeName
+ */
+function dt2jsCLI (ramlFile, ramlTypeName) {
+  dt2js.dt2js(ramlFile, ramlTypeName, writeToConsole)
```
Comment (postatum) [inline src/dt2js_cli.js]: Fixed https://github.com/raml-org/ramldt2jsonschema/commit/cdd565d8f005a61a701dca2d1869da79fff9e7dc

File: examples/js2dt_example.js [hunk 8]
```diff
@@ -0,0 +1,11 @@
+'use strict'
+var raml2json = require('ramldt2jsonschema')
+
```
Comment (postatum) [inline examples/js2dt_example.js]: Fixed https://github.com/raml-org/ramldt2jsonschema/commit/e9f20f36909f897d608565ec24b8f48bd3379a8e

File: src/js2dt_cli.js [hunk 9]
```diff
@@ -0,0 +1,38 @@
...
+  console.log(raml)
+}
+
+/**
+ * Just call js2dt.
+ *
+ * @param  {string} jsonFile
+ * @param  {string} ramlTypeName
+ */
+function js2dtCLI (jsonFile, ramlTypeName) {
```
Comment (postatum) [inline src/js2dt_cli.js]: Fixed https://github.com/raml-org/ramldt2jsonschema/commit/e9f20f36909f897d608565ec24b8f48bd3379a8e

Passing JSON as string so input type is identical in dt2js and js2dt - string containing file content.

File: src/dt2js.js [hunk 10]
```diff
@@ -0,0 +1,228 @@
...
+  return accum
+}
+
+/**
+ * Change RAML type of data to valid JSON schema type.
+ *
+ * @param  {Object} data
+ * @returns  {Object}
+ */
+function changeType (data) {
```
Comment (postatum) [inline src/dt2js.js]: Fixed https://github.com/raml-org/ramldt2jsonschema/commit/821de8951217a26a91b941b6f13e67d018c3d848

File: src/js2dt.js (hunk 1 above)
Comment (postatum) [inline src/js2dt.js]: It was originally called `processDefinitions` but then it became part of new `processDefinitions` func so I just added `_`. Renamed in https://github.com/raml-org/ramldt2jsonschema/commit/821de8951217a26a91b941b6f13e67d018c3d848

File: src/js2dt.js [hunk 11]
```diff
@@ -0,0 +1,352 @@
...
+}
+
+/**
+ * Replace $ref in data with defined type name.
+ * Type presence in 'definitions' is not validated.
+ *
+ * @param  {Object} data - Data containing $ref.
+ * @returns  {Object}
+ */
+function replaceRef (data) {
```
Comment (postatum) [inline src/js2dt.js]: Fixed https://github.com/raml-org/ramldt2jsonschema/commit/821de8951217a26a91b941b6f13e67d018c3d848

File: src/js2dt.js [hunk 12]
```diff
@@ -0,0 +1,352 @@
...
+  if (data.anyOf) {
+    return 'anyOf'
+  } else if (data.allOf) {
+    return 'allOf'
+  } else if (data.oneOf) {
+    return 'oneOf'
+  }
+}
+
+function addCombinationsType (data, combsKey) {
```
Comment (postatum) [inline src/js2dt.js]: Fixed https://github.com/raml-org/ramldt2jsonschema/commit/821de8951217a26a91b941b6f13e67d018c3d848

File: src/utils.js [hunk 13]
```diff
@@ -0,0 +1,51 @@
...
+  return obj
+}
+
+/**
+ * Title string.
+ *
+ * @param  {String} str
+ * @returns  {string}
+ */
+function title (str) {
```
Comment (postatum) [inline src/utils.js]: Fixed https://github.com/raml-org/ramldt2jsonschema/commit/821de8951217a26a91b941b6f13e67d018c3d848

**Pull request merged**